network_analysis/nbconverted/2A_community_detection_planted_partition.py

```diff
@@ -0,0 +1,160 @@
+#!/usr/bin/env python
+# coding: utf-8
+
+# ## Run community detection experiments for planted partition model
+# 
+# This script runs community detection on the eADAGE correlation network using the [planted partition model](https://en.wikipedia.org/wiki/Stochastic_block_model#Special_cases). Here, we use the implementation in the [graph-tool package](https://graph-tool.skewed.de/static/doc/demos/inference/inference.html#assortative-community-structure).
+# 
+# Using the planted partition model rather than the full stochastic block model ensures that communities are assortative (i.e. more connections within than across communities), which is what we're looking for in this analysis.
+
+# In[1]:
+
+
+import os
+
+import numpy as np
+import pandas as pd
+import graph_tool.all as gt
+import matplotlib.pyplot as plt
+import matplotlib.patches as mpatches 
+import seaborn as sns
+
+from sample_nodes import (
+    sample_degree_matched,
+    sort_by_degree,
+)
+
+gt.seed_rng(1)
+np.random.seed(1)
+
+
+# In[2]:
+
+
+# relevant file paths
+data_dir = './data'
+processed_graph = os.path.join(data_dir, 'eadage_generic_graph_unsigned.gt')
+
+# script parameters
+NUM_NODE_SAMPLES = 1000 # number of degree-matched node samples for permutation test
+NUM_BINS = 100 # number of bins to divide nodes into, for sampling
+
+
+# In[3]:
+
+
+G = gt.load_graph(processed_graph)
+# make sure vertex/edge properties exist
+print(G)
+print(list(G.vp.keys()))
+print(list(G.ep.keys()))
+
+
+# In[4]:
+
+
+# community detection using planted partition model
+state = gt.PPBlockState(G)
+state.multiflip_mcmc_sweep(beta=np.inf, niter=1000)
+
+
+# In[5]:
+
+
+print('k = {}'.format(state.get_B()))
+
+
+# In[6]:
+
+
+state.draw()
+
+
+# In[7]:
+
+
+# get dataframe mapping Pa genes to communities
+def graph_state_to_df(G, state):
+    labels = state.get_blocks().get_array()
+    vs = G.get_vertices()
+    names = [G.vp['name'][v] for v in vs]
+    degrees = G.get_total_degrees(vs)
+    is_generic = [G.vp['generic'][v] for v in vs]
+    return pd.DataFrame({
+        'gene': names,
+        'label': labels,
+        'degree': degrees,
+        'is_generic': is_generic
+    })
+
+labels_df = graph_state_to_df(G, state)
+print(len(labels_df.label.unique()))
+labels_df.sort_values(by='degree', ascending=False).head()
+
+
+# In[8]:
+
+
+# simultaneously sort nodes and degrees by degree, ascending
```

Comment:
```suggestion
# simultaneously sort nodes and degrees by degree, descending
```

Reply:
I think these are ascending? It actually doesn't matter which direction as long as they're sorted, but the check [here](https://github.com/greenelab/generic-expression-patterns/blob/e67bbc9eecfa0c51a2389f8db7ad764a0e0d7568/network_analysis/sample_nodes.py#L75) is making sure the degrees are sorted from small to large (i.e. ascending, I think).

When I print them below I'm printing in descending order, so maybe that's what's confusing you. I'll change that to ascending order to match the sorting.

Reply:
Ah yup i was going off the print statement:)